### PR TITLE
Fibonacci optimized using binet's formula

### PR DESCRIPTION
**Pull request by Aayush2111**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/main/java/com/thealgorithms/dynamicprogramming/Fibonacci.java [hunk 1]
```diff
@@ -80,15 +80,6 @@ public static int fibBotUp(int n) {
      * @author Shoaib Rayeen (https://github.com/shoaibrayeen)
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/dynamicprogramming/Fibonacci.java:80]: There are comments about the method that was implemented as an optimized Fibonacci.

File: src/main/java/com/thealgorithms/dynamicprogramming/Fibonacci.java [hunk 2]
```diff
@@ -80,15 +80,6 @@ public static int fibBotUp(int n) {
...
-            return 0;
-        }
-        int prev = 0, res = 1, next;
-        for (int i = 2; i <= n; i++) {
-            next = prev + res;
-            prev = res;
-            res = next;
-        }
-        return res;
+        return (int) (Math.pow(((1 + Math.sqrt(5)) / 2), n) / Math.sqrt(5));
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/dynamicprogramming/Fibonacci.java:83]: make another methods `fibBinet(int n)` and put this there and also put comments prior to that about explaining the Binet or put a link where people can find them.

**Review by debasishbsws — CHANGES_REQUESTED** [review]
And I think as input became large output will not fit inside int anymore. think about it

**Comment by Aayush2111** [comment]
Can you review again?
is it okay?

**Review by debasishbsws — APPROVED** [review]
Looks good

**Pull request merged**